scilpy/segment/streamlines.py

```diff
@@ -2,56 +2,48 @@
 
 import itertools
 
+from dipy.io.stateful_tractogram import StatefulTractogram
 from dipy.tracking.metrics import length
-from dipy.tracking.streamline import (set_number_of_points,
-                                      transform_streamlines)
-from dipy.tracking._utils import _mapping_to_voxel
+from dipy.tracking.streamline import set_number_of_points
 from dipy.tracking.vox2track import _streamlines_in_mask
 from nibabel.affines import apply_affine
 import numpy as np
 
 
-def target_line_based(streamlines, target_mask, affine=None, include=True):
+def target_line_based(sft, target_mask):
+    sft.to_vox()
```

Comment:
Could you add a small description? Ex, explain why you don't need to return data_per_point and data_per_streamline?
Reason: only used in filter_grid_roi, which does the dealing with data_per_point.

Maybe then this function should be _target_line_based?